terraform/deploy_new_portscan_instance.sh

```diff
@@ -14,31 +14,53 @@ function usage {
     exit 1
 }
 
+function check_dependencies {
+    if [ -z "$(command -v terraform)" ] || \
+        [ -z "$(command -v aws)" ] || \
+        [ -z "$(command -v jq)" ]
+    then
+        echo "This script requires the following tools to run:"
+        echo "- terraform"
+        echo "- aws (AWS CLI)"
+        echo "- jq"
+        exit 1
+    fi
+}
+
 function redeploy_instances {
-    tf_args=("-var-file=\"$workspace.tfvar\"")
+    tf_args=()
+    # Get all portscan instance IDs as a JSON array of dicts in the form:
+    # {
+    #   index: instance_index,
+    #   id: instance_id
+    # }
+    # Any previously removed instances are ignored (.deposed_key == null)
```

Comment:
"previously removed" :arrow_right: "previously-removed"

Reply:
I *think* that since `previously` is an adverb I'm safe. Thank you for bringing up the suggestion though!

Reply:
Ah, you are correct.  I have learned something.